src/lib.rs

```diff
@@ -368,6 +381,32 @@ impl Index<usize> for FixedBitSet
     }
 }
 
+/// Return a FixedBitSet containing bits set to **true** for every item in the iterator, other bits
+/// are set to **false**. The length of the FixedBitSet will be at least as large as the largest
+/// item in the input iterator.
+impl FromIterator<usize> for FixedBitSet
+{
+    fn from_iter<I: IntoIterator<Item=usize>>(src: I) -> Self {
+        let iter = src.into_iter();
+        let growth_factor = 2;
+        let (_, hi) = iter.size_hint();
+        let mut fbs = FixedBitSet::with_capacity(hi.unwrap_or(0));
+        let mut max = 0;
+        for i in iter {
+            max = std::cmp::max(max, i);
+            if i >= fbs.len() {
+                let new_len = std::cmp::max(i+1, fbs.len() * growth_factor);
+                fbs.grow(new_len);
+            }
+            fbs.put(i);
+        }
+        if fbs.len() > max + 1 {
+            fbs.shrink(max + 1);
+        }
+        fbs
```

Comment:
It makes sense but I'd break it down into blocks instead? Get the next 32 bits into a block, then push it to the storage and continue?

That said, maybe even better to start with Extend, and implement FromIterator using extend. Extend has a small complication with starting from an uneven block

Reply:
Ok I see the confusion! I'm reading this too quickly, I'm sorry about that.

My mind was at `Extend<bool>` and `FromIterator<bool>`; growing the set by appending bits. That's a different feature. This is good as it is.